### ISSUE TITLE
persist categorization preferences

### ISSUE THREAD

**Issue by iloveitaly** [ActivityWatch/aw-server-rust#394]
Is there a reason we don't persist categorization preferences on the server side? I can't imagine why this wouldn't be preferred. Would be a great QOL improvement.

**Comment by ErikBjare** [comment]
Tried to find the tracking issue for this, but couldn't find it.

The plan is:
 - add a KV store to the databases
 - implement appropriate API endpoint
 - make aw-webui always check server settings first, and if not present, initialize with store browser `localstore` settings (upgrade) or defaults (init).

I found this:
 - Implemented API endpoint in aw-server-rust: https://github.com/ActivityWatch/aw-server-rust/blob/master/aw-server/src/endpoints/settings.rs
 - Old stale attempt at bringing a keyvalue store into aw-server: https://github.com/ActivityWatch/aw-server/pull/84

So basically, what's left is adding support to aw-server-python (should be easy, just need to get the db migration right, but I just did something similar in https://github.com/ActivityWatch/aw-server/pull/128), then just do the aw-webui part.

I agree it would be a great QOL improvement, I've just personally gotten a bit lazy as it's good enough for me.

**Contributions welcome!**

**Comment by PetbkA** [comment]
BTW, current warning about settings storage isn't quite correct:
> These settings are only saved in your browser and will not remain **if you switch browser**

Settings aren't even depend on _browser_ but rather on _cookies_, 'cause if you clear them for AW site - you'll lost your settings even without switching browser

**Comment by ErikBjare** [comment]
@PetbkA They are technically stored in localStorage, not cookies.

But yes, if the user clears the site data then it is lost.

**Comment by ErikBjare** [comment]
I finished this today, see https://github.com/ActivityWatch/aw-webui/pull/493

Might need some battle-testing in beta releases (releasing v0.12.3b11 later today), but seems pretty ready!

**Comment by ErikBjare** [comment]
Tracking issue was here: https://github.com/ActivityWatch/activitywatch/issues/348

**Comment by iloveitaly** [comment]
Woohoo! So exciting :)

**Comment by RokeJulianLockhart** [comment]
@ErikBjare, this issue is still cited in `0.12.1b5`'s AOSP GUI:

<p></p><img width="1153" height="1118" alt="Image" src="https://github.com/user-attachments/assets/34f55d1a-a5f4-4aac-aec2-047b010ad881" /><p></p>

**Comment by ErikBjare** [comment]
@RokeJulianLockhart The Android app is running an older version of the web UI and server.

Once it gets updated (might be a while), the notice will disappear.
